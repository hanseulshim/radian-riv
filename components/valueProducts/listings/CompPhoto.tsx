import React, { useState } from 'react'
import { PropertyInterface } from 'api'
import { formatNullString, formatPrice } from 'utils'
import Checkbox from 'components/common/Checkbox'
import PhotoPlay from './PhotoPlay'

interface Props {
  property: PropertyInterface
  view: string
  checkedProperties: string[]
  setCheckedProperties: React.Dispatch<React.SetStateAction<string[]>>
}

export default function CompPhoto({
  property,
  view,
  checkedProperties,
  setCheckedProperties
}: Props) {
  const [listingChecked, setListingChecked] = useState(false)
  const [showDetails, toggleShowDetails] = useState(true)
  const getImagePath = (): string => {
    let icon = ''
    icon =
      view === 'Under Contract'
        ? 'contract'
        : view === 'Sold'
        ? 'sold'
        : 'listed'
    return property.distressed ? `distressed-${icon}` : icon
  }

  const checked = checkedProperties.includes(property.listingNumber)

  return (
    <div className="comp-photo">
      <div className="info-row">
        <div className="icon-row">
          <img
            className="icon"
            src={`${process.env.baseUrl}/images/${getImagePath()}.svg`}
            alt="property type"
          />
          <span style={{ fontWeight: 'bold' }}>
            #{formatNullString(property.order)}
          </span>
          <span style={{ marginLeft: '10px' }}>
            {view === 'Under Contract'
              ? view
              : `${view} for ${formatNullString(
                  formatPrice(
                    view === 'Sold' ? property.soldPrice : property.listingPrice
                  )
                )}`}
          </span>
        </div>
        <div className="icon-row">
          <span style={{ marginRight: 10 }}>
            {formatNullString(property.targetDistance)} miles
          </span>
          <Checkbox
            label=""
            checked={checked}
            onChange={() => {
              if (!checked) {
                const arr = checkedProperties.slice()
                arr.push(property.listingNumber)
                setCheckedProperties(arr)
              } else {
                const index = checkedProperties.findIndex(
                  num => num === property.listingNumber
                )
                const arr = checkedProperties.slice()
                arr.splice(index, 1)
                setCheckedProperties(arr)
              }
            }}
            leftLabel
            small
          />
        </div>
      </div>
      <PhotoPlay photos={property.photos} />
      <div className="detail-container">
        <div className="address">{formatNullString(property.address)}</div>
        <div className="label-row">
          <div className="label">Subdivision:</div>
          <span>{formatNullString(property.subdivision)}</span>
        </div>
        <div className="label-row">
          <div className="label">School District:</div>
          <span>{formatNullString(property.schoolDistrict)}</span>
        </div>
        <div className="toggle-button-container">
          <span className="spacer" />
          <div className="toggle-button">
            <div
              onClick={() => toggleShowDetails(!showDetails)}
              className={`toggle${showDetails ? ' active' : ''}`}
            >
              Details
            </div>
            <div
              onClick={() => toggleShowDetails(!showDetails)}
              className={`toggle${!showDetails ? ' active' : ''}`}
            >
              Description
            </div>
          </div>
          <span className="spacer" />
        </div>
        {showDetails ? (
          <div className="column-container">
            <div className="column">
              <div className="label-row">
                <div className="label">Bed</div>
                <span>{formatNullString(property.bed)}</span>
              </div>
              <div className="label-row">
                <div className="label">Bath</div>
                <span>{formatNullString(property.bath)}</span>
              </div>
              <div className="label-row">
                <div className="label">A.G. SQFT</div>
                <span>{formatNullString(property.agSqft)}</span>
              </div>
              <div className="label-row">
                <div className="label">Total SQFT</div>
                <span>{formatNullString(property.totalSqft)}</span>
              </div>
              <div className="label-row">
                <div className="label">AVE SQFT</div>
                <span>{formatNullString(property.sqft)}</span>
              </div>
              <div className="label-row">
                <div className="label">Year Built</div>
                <span>{formatNullString(property.yearBuilt)}</span>
              </div>
              <div className="label-row">
                <div className="label">Basement</div>
                <span>{formatNullString(property.basement)}</span>
              </div>
              <div className="label-row">
                <div className="label">Pool</div>
                <span>{formatNullString(property.pool)}</span>
              </div>
              <div className="label-row">
                <div className="label">Waterfront</div>
                <span>{formatNullString(property.waterfront)}</span>
              </div>
            </div>
            <div className="column">
              <div className="label-row">
                <div className="label">Concessions</div>
                <span>
                  {formatNullString(formatPrice(property.concessions))}
                </span>
              </div>
              <div className="label-row">
                <div className="label">COE Date</div>
                <span>{formatNullString(property.coeDate)}</span>
              </div>
              <div className="label-row">
                <div className="label">List Date</div>
                <span>{formatNullString(property.listingDate)}</span>
              </div>
              <div className="label-row">
                <div className="label">ACT DOM</div>
                <span>{formatNullString(property.actDom)}</span>
              </div>
              <div className="label-row">
                <div className="label">TOT DOM</div>
                <span>{formatNullString(property.totDom)}</span>
              </div>
              <div className="label-row">
                <div className="label">List Price</div>
                <span>
                  {formatNullString(formatPrice(property.listingPrice))}
                </span>
              </div>
              <div className="label-row">
                <div className="label">Listing #</div>
                <span className="link">
                  {formatNullString(property.listingNumber)}
                </span>
              </div>
            </div>
          </div>
        ) : (
          <div className="description-container">{property.description}</div>
        )}
        <div className="listing-export">
          <Checkbox
            label="Add Listing Sheet to Export"
            checked={listingChecked}
            onChange={() => {
              setListingChecked(!listingChecked)
            }}
            small
          />
        </div>
      </div>
    </div>
  )
}
