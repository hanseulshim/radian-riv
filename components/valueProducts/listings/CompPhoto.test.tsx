import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CompPhoto from './CompPhoto'

describe('CompPhoto', () => {
  let component
  beforeAll(() => {
    const property = {
      id: null,
      address: null,
      subdivision: null,
      schoolDistrict: null,
      bed: null,
      bath: null,
      agSqft: null,
      totalSqft: null,
      aveSqft: null,
      year: null,
      basement: null,
      pool: null,
      waterfront: null,
      concessions: null,
      coeDate: null,
      listDate: null,
      actDom: null,
      totDom: null,
      listPrice: null,
      soldPrice: null,
      listingNumber: null,
      exportMls: null,
      distressed: null,
      targetDistance: null,
      description: null,
      photos: [],
      rank: null,
      checked: null
    }
    const props = {
      property,
      view: 'view',
      checkedProperties: [],
      setCheckedProperties: () => {}
    }
    component = shallow(<CompPhoto {...props} />)
  })
  it('has comp-photo container', () => {
    expect(component.find('.comp-photo')).toHaveLength(1)
  })
  it('has comp-photo content', () => {
    expect(component.find('.detail-container')).toHaveLength(1)
    expect(component.find('.column-container')).toHaveLength(1)
    expect(component.find('.listing-export')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
