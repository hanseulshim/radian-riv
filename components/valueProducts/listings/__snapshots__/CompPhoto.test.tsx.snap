// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompPhoto should match snapshot 1`] = `
<div
  className="comp-photo"
>
  <div
    className="info-row"
  >
    <div
      className="icon-row"
    >
      <img
        alt="property type"
        className="icon"
        src="undefined/images/listed.svg"
      />
      <span
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        #
        1
      </span>
      <span
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        view for $547,804
      </span>
    </div>
    <div
      className="icon-row"
    >
      <span>
        0.41
         miles
      </span>
      <Checkbox
        checked={false}
        label=""
        leftLabel={true}
        onChange={[Function]}
        small={true}
      />
    </div>
  </div>
  <div
    className="photo-container"
  >
    <div
      className="photo-wrapper"
    >
      <img
        alt="property"
        className="photo"
      />
    </div>
    <div
      className="info-row"
    >
      <span
        style={
          Object {
            "width": 70,
          }
        }
      >
        1
         of 
        0
      </span>
      <div
        className="info-row"
      >
        <img
          alt="left button"
          className="icon"
          onClick={[Function]}
          src="undefined/images/left.svg"
        />
        <img
          alt="play/pause"
          className="icon"
          onClick={[Function]}
          src="undefined/images/play.svg"
        />
        <img
          alt="right button"
          className="icon"
          onClick={[Function]}
          src="undefined/images/right.svg"
        />
      </div>
      <span
        className="icon-row"
        onClick={[Function]}
      >
        <img
          alt="external link"
          className="icon"
          src="undefined/images/external.svg"
        />
        <span>
          All Photos
        </span>
      </span>
    </div>
  </div>
  <div
    className="detail-container"
  >
    <div
      className="address"
    >
      62030 Mariam Parkways
    </div>
    <div
      className="label-row"
    >
      <div
        className="label"
      >
        Subdivision:
      </div>
      <span>
        Soft Table Village
      </span>
    </div>
    <div
      className="label-row"
    >
      <div
        className="label"
      >
        School District:
      </div>
      <span>
        Avon County Public Schools
      </span>
    </div>
    <div
      className="toggle-button-container"
    >
      <span
        className="spacer"
      />
      <div
        className="toggle-button"
      >
        <div
          className="toggle active"
          onClick={[Function]}
        >
          Details
        </div>
        <div
          className="toggle"
          onClick={[Function]}
        >
          Description
        </div>
      </div>
      <span
        className="spacer"
      />
    </div>
    <div
      className="column-container"
    >
      <div
        className="column"
      >
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Bed
          </div>
          <span>
            5
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Bath
          </div>
          <span>
            3
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            A.G. SQFT
          </div>
          <span>
            2205
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Total SQFT
          </div>
          <span>
            2043
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            AVE SQFT
          </div>
          <span>
            2168
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Year Built
          </div>
          <span>
            2006
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Basement
          </div>
          <span>
            Full
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Pool
          </div>
          <span>
            Community
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Waterfront
          </div>
          <span>
            Water Oriented: No
          </span>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Concessions
          </div>
          <span>
            $16,044
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            COE Date
          </div>
          <span>
            1/19/2020
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            List Date
          </div>
          <span>
            4/22/2020
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            ACT DOM
          </div>
          <span>
            239
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            TOT DOM
          </div>
          <span>
            116
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            List Price
          </div>
          <span>
            $547,804
          </span>
        </div>
        <div
          className="label-row"
        >
          <div
            className="label"
          >
            Listing #
          </div>
          <span
            className="link"
          >
            MDMC695755
          </span>
        </div>
      </div>
    </div>
    <div
      className="listing-export"
    >
      <Checkbox
        checked={false}
        label="Add Listing Sheet to Export"
        onChange={[Function]}
        small={true}
      />
    </div>
  </div>
</div>
`;
